### PR TITLE
feat(i18n): Add new languages

### PR DESCRIPTION
**Pull request by dq18**
### What
- added all languages from Crowdin to the app
- added a translation completion level and number display in settings

List of languages was modified from:
https://github.com/annexare/Countries/blob/main/packages/countries/src/data/languages.ts

### CODE REVIEW

File: src/i18n/data/languages.json [hunk 1]
```diff
@@ -0,0 +1,937 @@
+[
+  {
+    "code": "aa",
+    "name": "Afar",
+    "native": "Afar"
+  },
+  {
+    "code": "ab",
+    "name": "Abkhazian",
+    "native": "\u0410\u04a7\u0441\u0443\u0430"
```
Comment (raphodn) [inline src/i18n/data/languages.json:10]: why the special characters ? not possible to avoid them ?

File: src/i18n/localeManager.js [hunk 2]
```diff
@@ -78,8 +79,66 @@ const localeManager = {
...
+        const localeCode = key.replace('./locales/', '').replace('.json', '')
+        return localeCode;
+      })
+  
+      return locales;
+    } catch (error) {
+      console.error('Error fetching locales:', error);
+    }
+  },
+  async calculateTranslationCompletion(locale) {
```
Comment (raphodn) [inline src/i18n/localeManager.js:97]: complex method, needs a description / docstring

File: src/views/UserSettings.vue [hunk 3]
```diff
@@ -69,23 +73,41 @@ export default {
...
 
+  },
   computed: {
     ...mapStores(useAppStore),
     formFilled() {
       return Object.values(this.userSettingsForm).every(x => !!x)
     },
   },
-  mounted() {
+  async mounted() {
```
Comment (raphodn) [inline src/views/UserSettings.vue:97]: "async" isn't used anywhere, are you sure we need it here ?

File: src/views/UserSettings.vue [hunk 4]
```diff
@@ -69,23 +73,41 @@ export default {
...
+      languageList: [],
+      languagesData: null,
+      languageTranslationCompletion: null,
     }
   },
+  watch:{
+    'userSettingsForm.selectedLanguage': async function () {
+      console.log("selected language: ", this.userSettingsForm.selectedLanguage)
+      if (this.userSettingsForm.selectedLanguage !== null) {
+        this.languageTranslationCompletion = await localeManager.calculateTranslationCompletion(this.userSettingsForm.selectedLanguage.code)
```
Comment (raphodn) [inline src/views/UserSettings.vue:85]: in the v2 it could be a github action instead (that stores the completion value in the `languages.json`)

**Review by raphodn — APPROVED** [review]
looks good ! don't forget to cleanup the console.logs :)

File: src/views/UserSettings.vue (hunk 4 above)
Comment (dq18) [reply src/views/UserSettings.vue:85]: Yes definitely

File: src/views/UserSettings.vue (hunk 3 above)
Comment (dq18) [reply src/views/UserSettings.vue:97]: in line 99 I put
`    const { default: languagesData } = await import('@/i18n/data/languages.json')
`

File: src/i18n/data/languages.json (hunk 1 above)
Comment (dq18) [reply src/i18n/data/languages.json]: I updated the file (formatting issue on Windows)

File: src/i18n/localeManager.js (hunk 2 above)
Comment (dq18) [reply src/i18n/localeManager.js:133]: Done

**Pull request merged**